spring-boot-crud-kotlin/src/test/kotlin/com/baeldung/controllers/TaskCRUDIntegrationTest.kt

```diff
@@ -0,0 +1,56 @@
+package com.baeldung.controllers
+
+import com.baeldung.crud.SpringBootCrudApplication
+import com.baeldung.crud.model.Task
+import com.baeldung.crud.model.TaskData
+import com.baeldung.crud.repositories.TaskEntity
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.client.TestRestTemplate
+
+@SpringBootTest(
+    classes = [SpringBootCrudApplication::class],
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT
+)
+class TaskCRUDIntegrationTest(
+    @Autowired var restTemplate: TestRestTemplate
+) {
+
+    var taskId: Long = 0
+
+    @Test
+    fun createTask() {
+        val taskData = TaskData("Task", "description", false)
+        val result = this.restTemplate.postForEntity("/tasks/create", taskData, TaskEntity::class.java)
+        taskId = result?.body?.id!!
+        assertTrue { result?.body?.name.equals("Task") }
+        assertTrue { result?.body?.description.equals("description") }
```

Comment:
I think the safe call on result is not needed so you can, do the dot call directly:

```suggestion
        assertTrue { result.body?.name.equals("Task") }
        assertTrue { result.body?.description.equals("description") }
```